File: jsp_servlet_exerc04/shopping-cart/src/main/java/cn/uncleyumo/shoppingcart/bean/Goods.java
package cn.uncleyumo.shoppingcart.bean;

import java.io.Serializable;

/**
 * @author uncle_yumo
 * @fileName Goods
 * @createDate 2024/11/5 November
 * @school 无锡学院
 * @studentID 22344131
 * @description
 */

public class Goods implements Serializable {
    private static final long serialVersionUID = 1L;
    private int gid;
    private String gname;
    private double gprice;
    private int gcount;

    public int getGid() {
        return gid;
    }

    public void setGid(int gid) {
        this.gid = gid;
    }

    public String getGname() {
        return gname;
    }

    public void setGname(String gname) {
        this.gname = gname;
    }

    public double getGprice() {
        return gprice;
    }

    public void setGprice(double gprice) {
        this.gprice = gprice;
    }

    public int getGcount() {
        return gcount;
    }

    public void setGcount(int gcount) {
        this.gcount = gcount;
    }
}

File: jsp_servlet_exerc03/src/main/kotlin/com/example/bean/Goods.kt
package com.example.bean

import java.io.Serializable

/**
 * @author uncle_yumo
 * @fileName Goods
 * @createDate 2024/10/23 October
 * @school 无锡学院
 * @studentID 22344131
 * @description
 */

class Goods(
    private var gid: Int?,
    private var gname: String?,
    private var gprice: Double?,
    private var gcount: Int?
) : Serializable {  // 实现Serializable接口，使其可以被序列化
    companion object {
        const val serialVersionUID = 1L
        fun getSerialVersionUID() = serialVersionUID
    }
    // 构造函数
    constructor() : this(null, null, null, null)
    constructor(gid: Int, gname: String, gprice: Double) : this(gid, gname, gprice, 0)
    // getter方法
    fun getGid() = gid
    fun getGname() = gname
    fun getGprice() = gprice
    fun getGcount() = gcount
    // setter方法
    fun setGid(gid: Int) {
        this.gid = gid
    }
    fun setGname(gname: String) {
        this.gname = gname
    }
    fun setGprice(gprice: Double) {
        this.gprice = gprice
    }
    fun setGcount(gcount: Int) {
        this.gcount = gcount
    }
}